src/overlays/actors/ovl_Dm_Bal/z_dm_bal.c

```diff
@@ -30,31 +36,178 @@ const ActorInit Dm_Bal_InitVars = {
+    f32 temp_fv1;
+    f32 temp_fv1_2;
+
+    this->unk_338 += 0x320;
+    this->unk_33A += 0x3E8;
+    temp_fv1 = (Math_CosS(this->unk_338) * 0.1f) + 1.0f;
+    this->unk_194 = temp_fv1;
+    this->unk_190 = temp_fv1;
+    temp_fv1_2 = (Math_SinS(this->unk_338) * 0.1f) + 1.0f;
+    this->unk_18C = temp_fv1_2 * temp_fv1_2;
+    this->actor.world.pos.y = (Math_SinS(this->unk_338) * 25.0f) + this->actor.home.pos.y;
+}
+
+void func_80C1ED0C(DmBal* this) {
+    if (this->unk_336 == 1) {
+        this->unk_198 = 1;
+    } else if (this->unk_334 >= 4) {
+        this->unk_334--;
+    } else if (this->unk_334 != 0) {
+        this->unk_198 = 1;
+        this->unk_334--;
+    } else {
+        this->unk_198 = 0;
+        this->unk_334 = 60;
+    }
+}
+
+void func_80C1ED64(DmBal* this, GlobalContext* globalCtx, Vec3f* arg2, Vec3f* arg3, f32 arg4) {
+    Actor* paper = Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_PAPER, arg2->x, arg2->y, arg2->z, 0, 0, 0, 0);
+
+    if (paper != NULL) {
+        paper->velocity = *arg3;
+        paper->gravity = arg4;
+    }
+}
+
+void DmBal_Update(Actor* thisx, GlobalContext* globalCtx) {
+    DmBal* this = THIS;
+    s32 pad;
+    Vec3f sp3C;
+    Vec3f sp30;
+
+    if (Animation_OnFrame(&this->skelAnime, 29.0f) && (this->skelAnime.animation == &object_bal_Anim_001804)) {
+        sp3C = this->actor.world.pos;
+        sp30 = D_80C1F2C4;
+        sp3C.x += 7.0f * Math_SinS(this->actor.shape.rot.y);
+        sp3C.y += 2.5f;
+        sp3C.z += 7.0f * Math_CosS(this->actor.shape.rot.y);
+        sp30.x = Math_SinS(this->actor.shape.rot.y) * 5.0f;
+        sp30.z = Math_CosS(this->actor.shape.rot.y) * 5.0f;
+        func_80C1ED64(this, globalCtx, &sp3C, &sp30, -0.4f);
+        func_80C1ED64(this, globalCtx, &sp3C, &sp30, -0.5f);
+    }
+    this->actionFunc(this, globalCtx);
+    func_80C1EAE8(this, globalCtx);
+    func_80C1EC60(this, globalCtx);
+    func_80C1ED0C(this);
+    SkelAnime_Update(&this->skelAnime);
+}
+
+s32 DmBal_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    DmBal* this = (DmBal*)THIS;
+    Vec3s rots;
+
+    if (limbIndex == 6) {
+        rots.x = Math_SinS(this->unk_33A) * 3640.0f;
+        rots.z = Math_CosS(this->unk_33A) * 3640.0f;
+        Matrix_RotateZYX(rots.x, 0, rots.z, MTXMODE_APPLY);
+        Matrix_Scale(this->unk_18C, this->unk_190, this->unk_194, MTXMODE_APPLY);
+        Matrix_RotateZS(-rots.z, MTXMODE_APPLY);
+        Matrix_RotateXS(-rots.x, MTXMODE_APPLY);
+    }
+    return false;
+}
+
+void DmBal_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+}
+
+void DmBal_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    DmBal* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+    func_8012C28C(globalCtx->state.gfxCtx);
+    gSPSegment(POLY_OPA_DISP++, 0x08, SEGMENTED_TO_VIRTUAL(D_80C1F2D0[this->unk_198]));
```

Comment:
Turns out from looking at the textures that these are eye textures, so you can name the array `sEyeTextures` and `unk_198` `eyeIndex`.